reana_job_controller/config.py

```diff
@@ -66,8 +69,13 @@
 ``/usr/local/share/mydata`` in the host machine.
 """
 
-SUPPORTED_COMPUTE_BACKENDS = os.getenv('COMPUTE_BACKENDS',
-                                       DEFAULT_COMPUTE_BACKEND).split(",")
+# How is this set in the environment?
+# It is hardcoded in Dockerfile and there is no code
+# in workflow-controller to override that for the job-controller.
+#SUPPORTED_COMPUTE_BACKENDS = os.getenv('COMPUTE_BACKENDS',
+#                                       DEFAULT_COMPUTE_BACKEND).split(",")
```

Comment:
Very good point for improvement 👍. The way this is set is a bit shady right now as one could pass it in REANA-Workflow-Controller with the process described in https://github.com/reanahub/reana-job-controller/pull/251/files#r416428666, but actually this is set at build time because [that knowledge is used to selectively install the correct packages](https://github.com/reanahub/reana-job-controller/blob/35c4fc8ab3c79a78e8709475e80ef2fd6b80bb85/Dockerfile#L23-L43).

So when we build images for `reana.cern.ch` we use the [Makefile](https://github.com/reanahub/reana/blob/master/Makefile) like:
```console
BUILD_ARGUMENTS="COMPUTE_BACKENDS=htcondorcern,slurmcern,kubernetes" BUILD_TYPE=release make build
```

What is happening behind the scenes is that we pass these [build args](https://docs.docker.com/engine/reference/commandline/build/#set-build-time-variables---build-arg) to REANA-Job-Controller `docker build`, which results in the final image having this environment variable.

Side note, as it might be of interest here: Then we take this image and we name it [`reanahub/reana-job-controller-htcondorcern-slurmcern`](https://hub.docker.com/r/reanahub/reana-job-controller-htcondorcern-slurmcern), not very pretty but this is what we have for now :) (see previous discussions [here](https://github.com/reanahub/reana/pull/289#issuecomment-612875297)).

Reply:
Thanks for the note Diego, it's very helpful! We'll be sure to follow that from now on. Perhaps we've been 'cheating' while building our images, we have simply been building them straight from the `dockerfile` within the RJC repo rather than through the `reanahub/reana` repo. I suppose this is a bad habit of ours.